pubtools/_content_gateway/cgw_client.py

```diff
@@ -0,0 +1,180 @@
+import json
+from .cgw_session import CGWSession
+
+class CGWClientError(Exception):
+    """
+    Custom exception to handle Content Gateway Client errors
+    """
+
+    def __init__(self, message):
+        self.message = message
+
+
+class CGWClient:
+    def __init__(self, hostname, cgw_auth=None, verify=True):
+        self.hostname = hostname.rstrip("/")
+        self.cgw_session = CGWSession(self.hostname, verify=verify)
+        if cgw_auth:
+            cgw_auth.make_auth(self.cgw_session)
+
+    def call_cgw_api(self, method, endpoint, params=None, data=None):
+        # Check if CGW hostname is present
+        if not self.hostname:
+            raise CGWClientError("No content gateway hostname found")
+
+        # Check if correct method id passed
+        if method not in ["GET", "POST", "PATCH", "PUT", "DELETE"]:
+            raise CGWClientError("Wrong request method passed")
+
+        response = None
+        output = {}
+        try:
+            if method == "GET":
+                response = self.cgw_session.get(endpoint, params=params)
+            elif method == "POST":
+                response = self.cgw_session.post(endpoint, data=data)
+            elif method == "PUT":
+                response = self.cgw_session.put(endpoint, data=data)
```

Comment:
There seems to be some inconsistency on where you pass extra params

Reply:
Didn't get you.